appinventor/blocklyeditor/src/generators/rules.js

```diff
@@ -87,6 +87,8 @@ Blockly.Rules['logic_backward_rule'] = function() {
 Blockly.Rules['logic_triple_pattern'] = function() {
   var subject = Blockly.Rules.valueToCode(this, 'SUBJECT', Blockly.Rules.ORDER_NONE) || Blockly.Rules.RULES_EMTPY_URI;
   var predicate = Blockly.Rules.valueToCode(this, 'PREDICATE', Blockly.Rules.ORDER_NONE) || Blockly.Rules.RULES_EMTPY_URI;
+  if (predicate == "a")
```

Comment:
I remember the generated rules not working for me. I did a quick check on desktop with jena 2.10.0 and it only works with "rdf:type".

(check the (informal) spec [here](https://jena.apache.org/documentation/inference/#RULEsyntax))

Reply:
Rather than branching here, we should change the rule generator for `logic_rdf_type`  to return `rdf:type` rather than `a`

Reply:
sure, no problem

(initially put it there so it could be generalized to any occurrence of "a", for instance, in a URI (since rdf_type doesn't have a visible dedicated block) but then it'd rather be "<a>" I suppose)